src/routes/shadowBanUser.ts

```diff
@@ -0,0 +1,89 @@
+import {db, privateDB} from '../databases/databases';
+import {getHash} from '../utils/getHash';
+import {Request, Response} from 'express';
+
+
+export async function shadowBanUser(req: Request, res: Response) {
+    const userID = req.query.userID as string;
+    const hashedIP = req.query.hashedIP as string;
+    let adminUserIDInput = req.query.adminUserID as string;
+
+    const enabled = req.query.enabled === undefined
+        ? false
+        : req.query.enabled === 'true';
+
+    //if enabled is false and the old submissions should be made visible again
+    const unHideOldSubmissions = req.query.unHideOldSubmissions !== "false";
+
+    if (adminUserIDInput == undefined || (userID == undefined && hashedIP == undefined)) {
+        //invalid request
+        res.sendStatus(400);
+        return;
+    }
+
+    //hash the userID
+    adminUserIDInput = getHash(adminUserIDInput);
+
+    const isVIP = db.prepare("get", "SELECT count(*) as userCount FROM vipUsers WHERE userID = ?", [adminUserIDInput]).userCount > 0;
+    if (!isVIP) {
+        //not authorized
+        res.sendStatus(403);
+        return;
+    }
+
+    if (userID) {
+        //check to see if this user is already shadowbanned
+        const row = privateDB.prepare('get', "SELECT count(*) as userCount FROM shadowBannedUsers WHERE userID = ?", [userID]);
+
+        if (enabled && row.userCount == 0) {
+            //add them to the shadow ban list
+
+            //add it to the table
+            privateDB.prepare('run', "INSERT INTO shadowBannedUsers VALUES(?)", [userID]);
+
+            //find all previous submissions and hide them
+            if (unHideOldSubmissions) {
+                db.prepare('run', "UPDATE sponsorTimes SET shadowHidden = 1 WHERE userID = ?", [userID]);
+            }
+        } else if (!enabled && row.userCount > 0) {
+            //remove them from the shadow ban list
+            privateDB.prepare('run', "DELETE FROM shadowBannedUsers WHERE userID = ?", [userID]);
+
+            //find all previous submissions and unhide them
+            if (unHideOldSubmissions) {
+                db.prepare('run', "UPDATE sponsorTimes SET shadowHidden = 0 WHERE userID = ?", [userID]);
+            }
+        }
+    }
+    // TODO TYPESCRIPT a lof unused code, wtf is happening
```

Comment:
what about all this commented out code?

Reply:
This is unfinished IP banning. It only will remove previous submissions right now but won't prevent new ones.